### PR TITLE
Employee search completed with pagination

### PR DESCRIPTION
**Pull request by Shwetakale**

### CODE REVIEW

File: app/controllers/office_automation_employee/users_controller.rb [hunk 1]
```diff
@@ -30,7 +30,18 @@ def update
...
       @company = Company.find params[:company_id]
+
+      if params[:q]
+        @users = @company.users.full_text_search(params[:q])
+        flash[:danger] = 'No result found' if @users.count == 0
+      else
+        @users = @company.users 
+      end
+
+      @users = @users.page(params[:page]).per(1) if @users.count > 0
```
Comment (kirandc) [inline app/controllers/office_automation_employee/users_controller.rb]: Why you using condition for pagination and .per(1) you set pagination 1 per page?

File: app/views/office_automation_employee/users/index.html.haml [hunk 2]
```diff
@@ -26,3 +39,4 @@
                   %td= user.email
                   %td= user.roles.to_sentence
                   %td= user.status if can? :manage, user
+          = paginate @users if @users.count > 0
```
Comment (kirandc) [inline app/views/office_automation_employee/users/index.html.haml]: Same here why used condition if @users.count > 0  for pagination show. No need it's If no result more than page count it's not show pagination

File: Gemfile [hunk 3]
```diff
@@ -24,6 +24,8 @@ gem 'carrierwave-mongoid', '~> 0.6.3', :require => 'carrierwave/mongoid'
 gem 'rmagick'
 gem 'sinatra', require: false
 gem 'colorbox-rails'
+gem 'mongoid_search'
```
Comment (kirandc) [inline Gemfile:27]: You added mongoid search let us know how it's exactly work.

Comment (Shwetakale) [inline Gemfile:27]: When we use mongoid_search we need to include Mongoid::Search in our model and we need to specify for which fields we want to search with the help of search_in. When we specify fields in search_in it adds _keyword field in document and when i created object from console _keyword array was consisting of keywords ex email is abc@gmail.com then it had keywords abc, gmail, com.. when we request comes for search it searches in keywords as i used full_text_search method if it finds complete word in _keywords it returns that object

File: app/views/kaminari/_gap.html.haml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+-#  Non-link tag that stands for skipped pages...
```
Comment (kirandc) [inline app/views/kaminari/_gap.html.haml:1]: Why this app/views/kaminari  pages needed?

File: app/mailers/office_automation_employee/user_mailer.rb [hunk 5]
```diff
@@ -7,10 +7,11 @@ def notification_email(company , user, updated_attributes)
       @user = user
       @company = company
       @admins = @company.users.where(:roles.in => [Role::ADMIN])
+      recipients = []
       @admins.each do |admin|
-
-        mail(to: admin.email, subject: "#{@user.profile.first_name} #{@user.profile.last_name} has updated his/her profile")
+        recipients << admin.email
```
Comment (kirandc) [inline app/mailers/office_automation_employee/user_mailer.rb]: why recipients array. You can directly use admin.email

Comment (kirandc) [inline app/mailers/office_automation_employee/user_mailer.rb]: recipients = @company.users.where(:roles.in => [Role::ADMIN]).map(&:email)

File: app/views/kaminari/_gap.html.haml (hunk 4 above)
Comment (Shwetakale) [inline app/views/kaminari/_gap.html.haml:1]: Sir actually kaminari has default views which had first and last and we dont want that so generated views using rails g kaminari:views default -e haml command and it was not supported for bootstrap3 so edited that..

**Pull request merged**